---
title: "Prolactin and its receptor: H-P-C gene expression analyses"
author: "Victoria"
date: "updated 9/11/2019"
output: html_document
---

```{r data, message=FALSE, warning=FALSE, include=FALSE}
library(tidyverse)
library(officer)#creates Ppt slides
library(rvg)#sends ggplot to ppt
library(emmeans) #post-hoc analyses
library(cowplot)
library(robust) #robust regression (lmRob)

prl<-read.csv(file="~/Desktop/data/PRL_PRLR_RNAseq_data.csv")

prl.long<-prl %>% gather(key="bird",value = "counts",4:991, factor_key = TRUE) #wide to long
prl.long$TranscriptID<-NULL #removes meaningless columns
prl.long$Num<-NULL 
prl.long<-separate(data = prl.long, col = bird, into = c("bird_id", "sex", "tissue","stage"), sep = "_") #splits bird details into separate columns by _ separator
prl.long$counts<-as.numeric(prl.long$counts)
prl.long$stage<-as.factor(prl.long$stage)
prl.long$manip <- ifelse(grepl("m", prl.long$stage), "y", "n") #y = yes, manipulation ; n=no,not a manip, a characterization

prl.long$stage <- factor(prl.long$stage, levels=c("control", "bldg", "lay", "inc.d3", "m.inc.d3","inc.d9", "m.inc.d8", "m.inc.d9", "inc.d17", "m.inc.d17", "hatch", "extend", "extend.hatch", "prolong", "inc.prolong", "m.hatch", "n5", "n9")) #manually re-orders stages.

prl.long$timept<-prl.long$stage #create numerical "timept" variable for ordering stages
levels(prl.long$timept)<-c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18") #rename levels in bulk! saves lines of code. 
prl.long$timept<-as.numeric(prl.long$timept) #continuous, numerical variable for x axis

#focal variable indicates time points I care about specifically 
prl.long$focal<-prl.long$timept
prl.long$focal<-ifelse(prl.long$stage %in% c("bldg", "inc.d3", "m.inc.d8", "inc.d9", "hatch"), "y", "n") #if the stage is in the list of accepted values, focal is yes, otherwise, no. 

prl.long$log.exp<-log(prl.long$counts) #log transform counts data 
```

###Crop sac qPCR data  
Some notes on this dataset: 
Genes-of-interest PRL and PRLR were run in *duplicate* for each crop sample, along with two reference genes, ACTB and rpL4. 
Gene expression was normalized (dCT) by subtracting the geometric mean of the reference gene duplicates (ACTB and rpL4) from the average Ct of the duplicates for each gene-of-interest.  
*The more positive the dCT, the lower the gene expression relative to reference genes (within a sample)*  
*Can also think of it as "distance from reference gene"*  
The second normalization (ddCT) involved subtracting the average normalized expression (dCT) of the control group (here, **building**) from each normalized sample for that gene. 
*More positive numbers imply lower expression than average, more negative numbers imply higher expression than average.*  
*For this reason, it is easier to interpret the -ddCT. Now, postive numbers on -ddCT imply higher expression than average control group, and negative numbers imply lower expression than average control group, as one would expect*  
Lowest numbers have lowest relative expression, highest numbers have highest relative expression.  
```{r crop data import, message=FALSE, warning=FALSE, include=FALSE}
crops<-read.csv(file="~/Desktop/data/2019-09-11_PRLR_project_results.csv")
myvars<-c("ID","ID_correct","Stage","Sex","PRL_ddCT", "PRLR_ddCT") #variables I want to keep from csv
crops<-crops[myvars] #subsets dataframe
crops.long<-crops %>% gather(key="gene", value="ddCT", 5:6, factor_key=TRUE)#wide to long
crops.long$Stage <- factor(crops.long$Stage, levels=c("Bldg", "Inc_D3", "Inc_D9","Manip_D8", "Hatch")) #manually re-orders stages.
crops.long$timept<-crops.long$Stage #create numerical "timept" variable for ordering stages
levels(crops.long$timept)<-c("1","2","3","4","5") #rename levels in bulk! saves lines of code.

#log transform
crops.long$negddCT<-(-(crops.long$ddCT))
range(crops.long$negddCT, na.rm=TRUE) #returns the range of neg ddCT values. Lowest is -7.7. 
#so, to get all values to be positive for log transformation, will add 8 to all and then log transform. 
crops.long$log.exp.neg<-log((crops.long$negddCT)+8)

#fold change for qPCR analysis
crops.long$foldchange<-2^(crops.long$negddCT)#fold change is 2^-(ddCt)
crops.long$log.foldchange<-log(crops.long$foldchange)
range(crops.long$foldchange, na.rm=TRUE)

#fold change distribution
hist.cr <- crops.long[crops.long$gene=="PRLR_ddCT",] #PRLR
hist(hist.cr$foldchange, main="Crop PRLR Fold Change")
hist(hist.cr$log.foldchange, main="Crop PRLR LOG Fold Change") #log fold change nice n normal now.

```

#Hypothalamus 
Transcriptomic data from RNAseq.   

##Data distribution  
*Note: "Focal data" refers to data from only the five stages we are interested in: bldg, inc d3, inc d9, m.inc d8, and hatch"*  
Histograms of log transformed data show  
HYP PRL: possible bimodal distribution,  
HYP PRLR: relatively normal distributon  
```{r hyp histograms, echo=FALSE, message=FALSE, warning=FALSE}
#baseR way
#subset data
hist.hp <- prl.long[prl.long$geneid=="prl" & prl.long$tissue=="hypothalamus" & prl.long$focal=="y",] #PRL
hist.hr <- prl.long[prl.long$geneid=="prlr" & prl.long$tissue=="hypothalamus" & prl.long$focal=="y",] #PRLR

#plot histograms
hist(hist.hp$log.exp, breaks=seq(3,7,0.1), main="Hyp. PRL (focal only)")
hist(hist.hr$log.exp, breaks=seq(3,7,0.1), main="Hyp. PRLR (focal only)")
```

##Gene expression plots
```{r hyp plots, echo=FALSE, message=FALSE, warning=FALSE}
bp.hyp.prl<- prl.long%>% filter(focal=="y" & tissue=="hypothalamus" & geneid=="prl") %>%
  ggplot(aes(y = log.exp, x = as.factor(timept)))+
  geom_point(aes(shape=sex, color=sex, fill=sex), position=position_jitter(width=0.1, height=0.1), size=3, alpha=1)+
  scale_shape_manual(values = c(21, 24))+ #shapes w/ color & fill, circle & triangle
  scale_color_manual(values=c("black","black"))+ #shape border color
  scale_fill_manual(values=c("gray50", "white"))+ #shape fill color
geom_boxplot(outlier.shape=NA, alpha=0)+ #no outliers because points plotted by geom_points
   scale_x_discrete(labels=c("building", "inc.d3", "inc.d9", "manip.d8",  "hatch"))+
  scale_y_continuous(limits=c(3,7))+ #added so that both HYP graphs have same y axis
  labs(x=NULL,y="log(estimated counts)", title="PRL")+
  theme_classic()+
  theme(text = element_text(size=16),
        plot.title = element_text(hjust = 0.05, size = 14), #slightly shifted to right 
        axis.text.x = element_text(angle=45, hjust=1))
print(bp.hyp.prl)

bp.hyp.prlr<- prl.long%>% filter(focal=="y" & tissue=="hypothalamus" & geneid=="prlr") %>%
  ggplot(aes(y = log.exp, x = as.factor(timept)))+
  geom_point(aes(shape=sex, color=sex, fill=sex), position=position_jitter(width=0.1, height=0.1), size=3, alpha=1)+
  scale_shape_manual(values = c(21, 24))+ #shapes w/ color & fill, circle & triangle
  scale_color_manual(values=c("black","black"))+ #shape border color
  scale_fill_manual(values=c("gray50", "white"))+ #shape fill color
  geom_boxplot(outlier.shape=NA, alpha=0.25)+
  scale_x_discrete(labels=c("building", "inc.d3", "inc.d9", "manip.d8",  "hatch"))+
  scale_y_continuous(limits=c(3,7))+ #added so that both HYP graphs have same y axis
  labs(x=NULL,y="log(estimated counts)", title="PRLR")+
  theme_classic()+
  theme(text = element_text(size=16),
        plot.title = element_text(hjust = 0.05,size=14),
        axis.text.x = element_text(angle=45, hjust=1))
print(bp.hyp.prlr)
```

```{r hyp figures, eval=FALSE, message=FALSE, warning=FALSE,echo=FALSE}
#only run once (eval set to false)
# paper-ready figures using cowplot
hyp.legend <- get_legend(bp.hyp.prl) #extract legend (both have same legend)
hyp.plots<-plot_grid(bp.hyp.prl + theme(legend.position="none"), #hides legedn on plot
                     bp.hyp.prlr + theme(axis.title.y = element_blank(), legend.position="none"), #hides y axis on 2nd plot and legend 
                     labels = c("A", "B"), #adds letters
                     align = "h") #place graphs side by side, align axes

hyp.plots <- plot_grid(hyp.plots, hyp.legend, rel_widths = c(3, .2))

save_plot("~/Desktop/hyp.plots.pdf", hyp.plots, #saves above object as filename 
          ncol = 3, #number of columns
          nrow = 1, #number of rows
          base_aspect_ratio = 1.1) #base aspect ratio of each individual plot
```
###GLM: Hyp PRL  
ANOVA indicates a significant effect of stage, but no effect of sex or any interaction.  
According to p-values in glm, inc.d3 and hatch significantly differs from manipulation. (But manip does not significantly differ from hatch).  

**Residual inspection**: Residual plots show some outliers with large Cook's distances and high influence.  Tails of Q-Q plot are not on the line.   
(data points: #64, 79, 85)
```{r lm hyp prl, echo=FALSE, message=FALSE, warning=FALSE}
#HYP PRL
hprl<-prl.long %>% filter(tissue=="hypothalamus" & geneid=="prl" & focal=="y")
m.hp.int<-glm(log.exp~stage*sex, data=hprl, family="gaussian")
anova(m.hp.int, test="F")#F test that gives similar results to likelihood ratio test, but now comparable to the robust F test from lmRob 
anova(m.hp.int, test="Chisq") #tests the influence of each factor in the model #likelihood ratio test
#because interaction not significant for this, reporting results of model w/o interaction
m.hp<- glm(log.exp~stage+sex, data=hprl, family="gaussian")

##Stage influencial to data, but not sex or interaction. 
summary(m.hp) #comparisons to female, building (default reference)

#post-hoc comparisons
# emm.hp<-emmeans(m.hp, "stage") #run emmeans pairwise by stage
# pairs(emm.hp) #p-values with pairwise comparisons
# CLD(emm.hp) #gives letters (in numbers) for a graph later on. 

plot(m.hp)
```

```{r lmRob hyp prl, echo=FALSE}
library(robust)
hp.rob<-lmRob(log.exp~stage+sex, data=hprl)
anova.lmRob(hp.rob) #significant effect of stage, none of sex
summary(hp.rob) #sig. effect of inc.d8, weak trend towards hatch

```

###GLM: Hyp PRLR
"ANOVA" indicates that there is a significant effect of stage and sex on PRLR expression, but no interaction.  
Manipulation day 8 is significantly higher than bldg, day 3, and day 9, but NOT significantly different from hatch.   
All other stages not signficantly different from each other.  
Significant sex difference, where males > females.   

**Residual inspection**: Outliers with large Cook's distance and influence (datapoints: #2, #87, #90)  

```{r lm hyp prlr, echo=FALSE, message=FALSE, warning=FALSE}
#HYP PRLR
hr<-prl.long %>% filter(tissue=="hypothalamus" & geneid=="prlr" & focal=="y")
m.hr.int<- glm(log.exp~stage*sex, data=hr, family="gaussian")
m.hr<-glm(log.exp~stage+sex, data=hr, family="gaussian")
anova(m.hr.int, test="F") 
anova(m.hr.int, test="Chisq") 
anova(m.hr, m.hr.int, test="Chisq") #likelihood ratio test
#interaction not significant / important for model, run downstream tests on model w/o interaction
summary(m.hr)

#post-hoc comparisons
# emm.hr<-emmeans(m.hr, "stage") #run emmeans pairwise by stage
# pairs(emm.hr) #p-values with pairwise comparisons
# CLD(emm.hr) #gives letters (in numbers) for a graph later on.

#this pattern is likely being driven by the outliers

plot(m.hr)
```

```{r lmRob hyp prlr, echo=FALSE}
hr.rob<-lmRob(log.exp~stage+sex, data=hr)
anova.lmRob(hr.rob) #significant effect of stage and of sex
summary(hr.rob)
```

#Pituitary  
Transcriptomic data from RNAseq.  

##Data distribution  
Histograms of log transformed data show  
PIT PRL: possible bimodal distribution, w/ one large outlier,  
PIT PRLR: relatively normal distributon  
```{r pit histograms, echo=FALSE, message=FALSE, warning=FALSE}
#baseR way
#subset data
hist.pp <- prl.long[prl.long$geneid=="prl" & prl.long$tissue=="pituitary" & prl.long$focal=="y",] #PRL
hist.pr <- prl.long[prl.long$geneid=="prlr" & prl.long$tissue=="pituitary" & prl.long$focal=="y",] #PRLR

#plot histograms
hist(hist.pp$log.exp, breaks=seq(1,14,0.1), main="Pit. PRL (focal only)")
hist(hist.pr$log.exp, breaks=seq(1,14,0.1), main="Pit. PRLR (focal only)")
```
##Gene exp. plots
```{r pit plots, echo=FALSE,message=FALSE, warning=FALSE}
#PIT PRL  
bp.pit.prl<- prl.long%>% filter(focal=="y" & tissue=="pituitary" & geneid=="prl") %>%
  ggplot(aes(y = log.exp, x = as.factor(timept)))+
  geom_point(aes(shape=sex, color=sex, fill=sex), position=position_jitter(width=0.1, height=0.1), size=3, alpha=1)+
  scale_shape_manual(values = c(21, 24))+ #shapes w/ color & fill, circle & triangle
  scale_color_manual(values=c("black","black"))+ #shape border color
  scale_fill_manual(values=c("gray50", "white"))+ #shape fill color
  geom_boxplot(outlier.shape=NA, alpha=0.25)+
  scale_y_continuous(limits=c(7,14))+
  scale_x_discrete(labels=c("building", "inc.d3", "inc.d9", "manip.d8",  "hatch"))+
  labs(x=NULL,y="log(estimated counts)", title="PRL")+
  theme_classic()+
  theme(text = element_text(size=16),
        plot.title = element_text(hjust = 0.05, size=14),
        axis.text.x = element_text(angle=45, hjust=1))

bp.pit.prlr<- prl.long%>% filter(focal=="y" & tissue=="pituitary" & geneid=="prlr") %>%
  ggplot(aes(y = log.exp, x = as.factor(timept)))+
  geom_point(aes(shape=sex, color=sex, fill=sex), position=position_jitter(width=0.1, height=0.1), size=3, alpha=1)+
  scale_shape_manual(values = c(21, 24))+ #shapes w/ color & fill, circle & triangle
  scale_color_manual(values=c("black","black"))+ #shape border color
  scale_fill_manual(values=c("gray50", "white"))+ #shape fill color
  geom_boxplot(outlier.shape=NA, alpha=0.25)+
    theme(axis.text=element_text(size=22),
        axis.title=element_text(size=24,face="bold"),
        plot.caption = element_text(size=10, hjust=0),
        legend.position="none")+
  scale_x_discrete(labels=c("building", "inc.d3", "inc.d9", "manip.d8",  "hatch"))+
  scale_y_continuous(limits=c(3,7))+ #same limits as HYP PRLR graph
  labs(x=NULL,y="log(estimated counts)", title="PRLR")+
  theme_classic()+
  theme(text = element_text(size=16),
        plot.title = element_text(hjust = 0.05, size=14),
        axis.text.x = element_text(angle=45, hjust=1))

print(bp.pit.prl)
print(bp.pit.prlr)
```
```{r pit figures, eval=FALSE, message=FALSE, warning=FALSE, echo=FALSE}
#only run once (eval set to false)
# paper-ready figures using cowplot
pit.legend <- get_legend(bp.pit.prl) #extract legend (both have same legend)
pit.plots<-plot_grid(bp.pit.prl + theme(legend.position="none"), #hides legedn on plot
                     bp.pit.prlr + theme(axis.title.y = element_blank(), legend.position="none"), #hides y axis on 2nd plot and legend 
                     labels = c("A", "B"), #adds letters
                     align = "h") #place graphs side by side, align axes

hyp.plots <- plot_grid(pit.plots, pit.legend, rel_widths = c(3, .2))

save_plot("~/Desktop/pit_plots.pdf", pit.plots, #saves above object as filename 
          ncol = 3, #number of columns
          nrow = 1, #number of rows
          base_aspect_ratio = 1.1) #base aspect ratio of each individual plot
```


###GLM: Pit PRL
"ANOVA" : significant effects of stage and sex, with no interaction.   
Manipulation day 8 is significantly higher than bldg, day 3, and day 9, but NOT significantly different from hatch.   
All other stages not signficantly different from each other.  
Significant sex difference, where males < females.   

**Residual inspection**: Outliers with large Cook's distance and influence (datapoints: #8, #28, #78)  

```{r lm pit prl, echo=FALSE, message=FALSE, warning=FALSE}
#PIT PRL
pp<-prl.long %>% filter(tissue=="pituitary" & geneid=="prl" & focal=="y")
m.pp.int<- glm(log.exp~stage*sex, data=pp, family="gaussian")
m.pp<- glm(log.exp~stage+sex, data=pp, family="gaussian")
anova(m.pp.int, test="F") 
anova(m.pp.int, test="Chisq") 
anova(m.pp.int, m.pp, test="Chisq") #weak trend towards affecting the interaction, but not present here. 

#summary and post-hoc analyses on stage+sex model
# summary(m.pp)
# emm.pp<-emmeans(m.pp, ~stage)
# pairs(emm.pp)
# CLD(emm.pp)

plot(m.pp)
```

```{r lmRob pit prl, echo=FALSE}
pp.rob<-lmRob(log.exp~stage+sex, data=pp)
anova.lmRob(pp.rob) #significant effect of stage and of sex
summary(pp.rob)
```

###GLM: Pit PRLR
"ANOVA" : significant effects of sex only 
Manipulation day 8 is significantly different from hatch and building, but not incubation stages.   
Significant sex difference, where males > females.   

**Residual inspection**: Outliers with large Cook's distance and influence (datapoints: #5, #18, #45)  

```{r lm pit prlr, echo=FALSE, message=FALSE, warning=FALSE}
#PIT PRLR
pr<-prl.long %>% filter(tissue=="pituitary" & geneid=="prlr" & focal=="y")
m.pr.int<- glm(log.exp~stage*sex, data=pr, family="gaussian")
m.pr<-glm(log.exp~stage+sex, data=pr, family="gaussian")
anova(m.pr.int, test="F") 
anova(m.pr.int, test="Chisq") 
anova(m.pr, m.pr.int, test="Chisq")
#interaction not significant

#summary and post-hoc with stage+sex model
# summary(m.pr)
# emm.pr<-emmeans(m.pr,"stage")
# pairs(emm.pr)
# CLD(emm.pr)

plot(m.pr)
```
```{r lmRob pit prlr, echo=FALSE}
pr.rob<-lmRob(log.exp~stage+sex, data=pr)
anova.lmRob(pr.rob) 
summary(pr.rob)
```

#Crop sac

##Data distribution  
Histograms of log transformed data show  
Crop PRL: *relatively* normal, with ~ 4 low outliers  
Crop PRLR: relatively normal distributon  
```{r crop histograms, echo=FALSE, message=FALSE, warning=FALSE}
#subset data
hist.cp <- crops.long[crops.long$gene=="PRL_ddCT",] #PRL
hist.cr <- crops.long[crops.long$gene=="PRLR_ddCT",] #PRLR

#plot histograms (-ddCT)
hist(hist.cp$negddCT, breaks=seq(-10,15,0.1), main="Crop PRL -ddCT") 
hist(hist.cr$negddCT, breaks=seq(-10,15,0.1), main="Crop PRLR -ddCT") #relatively normal

#histograms for log normalized data 
hist(hist.cp$log.exp.neg, breaks=seq(-5,5,0.1), main="Crop PRL log(-ddCT + 8)") 
hist(hist.cr$log.exp.neg, breaks=seq(-5,5,0.1), main="Crop PRLR log(-ddCT + 8)")
```

##Crop sac gene expression plots 

```{r crop plots, echo=FALSE, message=FALSE, warning=FALSE}
bp.crop.prl<- crops.long%>% filter(gene=="PRL_ddCT") %>%
  ggplot(aes(y = log.foldchange, x = as.factor(timept)))+
  geom_point(aes(shape=Sex, colour=Sex, fill=Sex), position=position_jitter(width=0.1, height=0.1), size=3, alpha=1)+
  scale_shape_manual(values = c(21, 24))+ #shapes w/ color & fill, circle & triangle
  scale_color_manual(values=c("black","black"))+ #shape border color
  scale_fill_manual(values=c("gray50", "white"))+ #shape fill color
  geom_boxplot(outlier.shape=NA, alpha=0.25)+
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"),
        plot.caption = element_text(size=10, hjust=0),
        legend.position="none")+
 scale_x_discrete(labels=c("building", "inc.d3", "inc.d9", "manip.d8",  "hatch"))+
  labs(x=NULL,y="log(fold change)", title="PRL")+
  scale_y_continuous(limits=c(-0.5,3))+ #same y axis between crop graphs
  theme_classic()+
  theme(text = element_text(size=16),
        plot.title = element_text(hjust = 0.05, size=14),
        axis.text.x = element_text(angle=45, hjust=1))

bp.crop.prlr<- crops.long%>% filter(gene=="PRLR_ddCT") %>%
  ggplot(aes(y = log.foldchange, x = as.factor(timept)))+
  geom_point(aes(shape=Sex, colour=Sex, fill=Sex), position=position_jitter(width=0.1, height=0.1), size=3, alpha=1)+
  scale_shape_manual(values = c(21, 24))+ #shapes w/ color & fill, circle & triangle
  scale_color_manual(values=c("black","black"))+ #shape border color
  scale_fill_manual(values=c("gray50", "white"))+ #shape fill color
  geom_boxplot(outlier.shape=NA, alpha=0.25)+
    theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"),
        plot.caption = element_text(size=10, hjust=0),
        legend.position="none")+
  scale_x_discrete(labels=c("building", "inc.d3", "inc.d9", "manip.d8",  "hatch"))+
  labs(x=NULL,y="log (fold change)", title="PRLR")+
  scale_y_continuous(limits=c(-0.5,3))+ #same y axis between crop graphs 
  theme_classic()+
  theme(text = element_text(size=16),
        plot.title = element_text(hjust = 0.05, size=14),
        axis.text.x = element_text(angle=45, hjust=1))

print(bp.crop.prl)
print(bp.crop.prlr)
```

```{r crop figures, eval=FALSE, message=FALSE, warning=FALSE, echo=FALSE}
#only run once (eval set to false)
# paper-ready figures using cowplot
crp.legend <- get_legend(bp.crop.prl) #extract legend (both have same legend)
crp.plots<-plot_grid(bp.crop.prl + theme(legend.position="none"), #hides legedn on plot
                     bp.crop.prlr + theme(axis.title.y = element_blank(), legend.position="none"), #hides y axis on 2nd plot and legend 
                     labels = c("A", "B"), #adds letters
                     align = "h") #place graphs side by side, align axes

crp.plots <- plot_grid(crp.plots, crp.legend, rel_widths = c(3, .2))

save_plot("~/Desktop/crop_plots.pdf", crp.plots, #saves above object as filename 
          ncol = 3, #number of columns
          nrow = 1, #number of rows
          base_aspect_ratio = 1.1) #base aspect ratio of each individual plot
```


###GLM: Crop PRL  
"ANOVA" : very weak trend towards an interaction, but not really anything else there. 
Not really anything going on in downstream analyses... 
Discuss mostly the fact that PRL gene expression was detectable in the crop.  

**Residual inspection**: Outliers with large Cook's distance and influence (datapoints: #29, #35, #85)  

```{r lm crop prl, echo=FALSE, message=FALSE, warning=FALSE}
#Crop PRL
cp<-crops.long %>% filter(gene=="PRL_ddCT")
m1.cp<- glm(log.exp.neg~Stage+Sex, data=cp, family="gaussian")
anova(m1.cp, test="F")

summary(m1.cp) #comparisons to female, building (default reference)

# emm.cp<-emmeans(m1.cp, ~Stage)
# pairs(emm.cp)
# CLD(emm.cp)

plot(m1.cp)
```

```{r lmRob crop prl, echo=FALSE}
cp.rob<-lmRob(log.exp.neg~Stage+Sex, data=cp)
anova.lmRob(cp.rob) #significant effect of stage and of sex
summary(cp.rob)
```

###PRLR Crop
"ANOVA" : weak trend towards stage
Not really anything going on in downstream comparisons. Trend towards hatch being significant (may be driving the weak trend towards an effect of stage)?  

**Residual inspection**: Outliers with large Cook's distance and influence (datapoints: #17, #62, #83), some bad outliers on this one!   

```{r lm crop prlr, echo=FALSE, message=FALSE, warning=FALSE}
#Crop PRLR
cr<-crops.long %>% filter(gene=="PRLR_ddCT")
m1.cr<- glm(log.exp.neg~Stage+Sex, data=cr, family="gaussian")
anova(m1.cr, test="Chisq") 
anova(m1.cr, test="F") 
summary(m1.cr) #comparisons to female, building (default reference)

# emm.cr<-emmeans(m1.cr, ~Stage)
# pairs(emm.cr)
# CLD(emm.cr)

plot(m1.cr)
```

```{r lmRob crop prlr, echo=FALSE}
cr.rob<-lmRob(log.exp.neg~Stage+Sex, data=cr)
anova.lmRob(cr.rob) #significant effect of stage ONLY, driven by hatch
summary(cr.rob)
```

```{r line plots comparing expression levels, echo=FALSE, eval=FALSE}
#how can you put these two graphs together?
all.prl<-prl.long %>% filter(focal=="y") %>%
  group_by(tissue,timept)%>%
  summarize(avg_exp=mean(log.exp, na.rm = TRUE)) %>%
  ggplot(aes(x=timept, y=avg_exp, color=tissue))+
  geom_line(size=1, alpha=1)
print(all.prl)


prl.long<-mutate(prl.long,
   tissxgene = paste(tissue, geneid,sep = ' ')) #creates the tissxgene value

all.genes<-prl.long %>% filter(focal=="y" & tissue!= "gonad") %>%
  group_by(tissxgene, timept)%>%
  summarize(avg_exp=mean(log.exp, na.rm = TRUE)) %>%
  ggplot(aes(x=timept, y=avg_exp, color=tissxgene, linetype=tissxgene))+
  geom_line(size=1, alpha=1)+
  scale_x_continuous(breaks=c(2,3,4,6,9,11), labels=c("bldg", "lay", "inc.d3", "m.inc.d8",  "inc.d17", "hatch"))+
  scale_linetype_manual(values=c("longdash", "solid","longdash","solid"))+
  scale_color_manual(values=c("turquoise2", "turquoise4","salmon1","salmon3"))+
  labs(y="log(estimated counts)", x="parental care stage")+
  theme_minimal()
print(all.genes + theme(legend.position="bottom"))
```
###Plasma prolactin correlations

```{r data prep for correlations, echo=FALSE, message = FALSE}
library(stringr)# for string manipulation 

#create a clean_id variable for band combinations in prl.long RNAseq dataset
prl.long$clean_id<-str_replace_all(prl.long$bird_id, "x", "") #remove meaningless "x" from band combos
prl.long$clean_id<-str_replace_all(prl.long$clean_id, "[[:punct:]]", "") #remove punctuation

#import plasma prolactin dataset
plasma<-read.csv("~/Desktop/data/prolactin_ria_data.csv") 

#create clean_id variable in that dataset 
plasma$clean_id<-str_replace_all(plasma$ColorBands, "x", "")
plasma$clean_id<-str_replace_all(plasma$clean_id,"[[:punct:]]", "" )

#join prolactin plasma data with prl.long RNAseq data
prl.join<-left_join(prl.long, plasma, by ="clean_id")

#table that returns the counts of entries that now have prolactin plasma data, sorted by stage and sex
#sample size is ok
prl.join %>%
  filter(focal=="y")%>%
  group_by(stage,sex)%>%
  drop_na(Prolactin.ng.mL) %>%
  summarise(n= n_distinct(Prolactin.ng.mL))
```
```{r}
#does plasma prolactin vary with stage in this dataset?

#bulk rename 
library(forcats)
plasma <- plasma %>%
    mutate(Treatment = fct_recode(Treatment,
                            "bldg" = "Bldg",
                            "inc_d17" = "Inc_d17",
                            "inc_d3" = "Inc_d3",
                            "inc_d9" = "Inc_d9",
                            "hatch" = "Hatch",
                            "M_inc8" = "M_Inc8"))
#note focal timepoints in prolactin dataset
plasma$focal<-ifelse(plasma$Treatment %in% c("bldg", "inc_d3", "M_inc8", "inc_d9", "hatch"), "y", "n") #if the stage is in the list of accepted values, focal is yes, otherwise, no. 

#manually re-order stages
plasma.focal$Treatment<- factor(plasma.focal$Treatment, levels=c("bldg", "inc_d3","inc_d9",  "M_inc8", "hatch")) 

#subset data
plasma.focal<-plasma %>% filter(focal=="y")

#linear model for sex and stage with plasma prolactin data
m.plasma<-glm(Prolactin.ng.mL~Treatment + Sex, plasma.focal, family="gaussian")
summary(m.plasma)
hist(m.plasma$residuals) #normal residuals :) 

#results from re-levels: 
#bldg sig from all but m_inc8
#inc_d3 sig from bldg and hatch
#inc_d9 sig from bldg and hatch
#m_inc8 sig from hatch only
#hatch sig from ALL.


#plot for plasma prolactin data 
bp.plasma<- plasma.focal %>%
  ggplot(aes(y = Prolactin.ng.mL, x = as.factor(Treatment)))+
  geom_point(aes(shape=Sex, colour=Sex, fill=Sex), position=position_jitter(width=0.1, height=0.1), size=3, alpha=1)+
  scale_shape_manual(values = c(21, 24))+ #shapes w/ color & fill, circle & triangle
  scale_color_manual(values=c("black","black"))+ #shape border color
  scale_fill_manual(values=c("gray50", "white"))+ #shape fill color
  geom_boxplot(outlier.shape=NA, alpha=0.25) +
   theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14,face="bold"),
        plot.caption = element_text(size=10, hjust=0),
        legend.position="none")+
 scale_x_discrete(labels=c("building", "inc.d3", "inc.d9", "manip.d8",  "hatch"))+
  labs(x=NULL,y="Plasma concentration (ng/mL)", title="Plasma prolactin levels")+
  scale_y_continuous(limits=c(0,100))+ #same y axis between crop graphs
  theme_classic()+
  theme(text = element_text(size=16),
        plot.title = element_text(hjust = 0.05, size=14),
        axis.text.x = element_text(angle=45, hjust=1))

bp.plasma
```

```{r proof of concept: prolactin vs pitPRL, echo = FALSE, message = FALSE}
#beginning with pit PRL versus plasma prolactin, as a proof of principle
#should be highly correlated 

#create a datasubset (pit, prl, join)
p.p.j<-prl.join%>% filter(focal=="y" & tissue=="pituitary" & geneid=="prl")

#plot correlation
#(scatterplot for hyp, prl, joined data) = sp.h.pr.j
sp.p.p.j<- p.p.j %>% ggplot(aes(y = log.exp, x = Prolactin.ng.mL, color=stage))+
           geom_point()+
          geom_smooth(method=lm, na.rm = TRUE, fullrange= TRUE,
              aes(group=1),colour="black") + 
          labs(x="Plasma prolactin concentration (ng/mL)", 
               y="Pituitary PRL expression (log(counts))")+
          ggtitle("Plasma prolactin predicts pituitary PRL expression", subtitle = "Proof of principle") + 
          annotate("text", x = 75, y = 3, label = "R2= 0.64, p<< 0.01")
sp.p.p.j

#correlation linear model 
corr.p.p<- lm(log.exp~Prolactin.ng.mL, p.p.j) #use lm as it gives R2 
summary(corr.p.p)  #incredibly significant! 
#SUCCESS!
```

```{r prolactin vs hyp PRLR, echo = FALSE, message = FALSE}
#create a datasubset (hyp, prlr, join)
h.pr.j<-prl.join%>% filter(focal=="y" & tissue=="hypothalamus" & geneid=="prlr")

#plot correlation
#(scatterplot for hyp, prlr, joined data) = sp.h.pr.j
sp.h.pr.j<- h.pr.j %>% ggplot(aes(y = log.exp, x = Prolactin.ng.mL, color=stage))+
           geom_point()+
          geom_smooth(method=lm, na.rm = TRUE, fullrange= TRUE,
              aes(group=1),colour="black") + 
          labs(x="Plasma prolactin concentration (ng/mL)", 
               y="Hypothalamic PRLR expression (log(counts))")+
          ggtitle("Prolactin vs Hyp PRLR") + 
          annotate("text", x = 75, y = 3, label = "R2=0.001, n.s.")
sp.h.pr.j

#correlation linear model 
corr.h.pr<- lm(log.exp~Prolactin.ng.mL, h.pr.j) #use lm as it gives R2 
summary(corr.h.pr)  #n.s.
```
```{r prolactin vs hyp PRL, echo = FALSE, message = FALSE}
#correlate plasma prolactin versus hyp ePRL exprssion 

#create a datasubset (hyp, prl, join)
h.p.j<-prl.join%>% filter(focal=="y" & tissue=="hypothalamus" & geneid=="prl")

#plot correlation
#(scatterplot for hyp, prl, joined data) = sp.h.pr.j
sp.h.p.j<- h.p.j %>% ggplot(aes(y = log.exp, x = Prolactin.ng.mL, color=stage))+
           geom_point()+
          geom_smooth(method=lm, na.rm = TRUE, fullrange= TRUE,
              aes(group=1),colour="black") + 
          labs(x="Plasma prolactin concentration (ng/mL)", 
               y="Hypothalamic PRL expression (log(counts))")+
          ggtitle("Plasma prolactin predicts hypothalamic ePRL expression") + 
          annotate("text", x = 75, y = 3, label = "R2=0.072, p = 0.02")
sp.h.p.j

#correlation linear model 
corr.h.p<- lm(log.exp~Prolactin.ng.mL, h.p.j) #use lm as it gives R2 
summary(corr.h.p)  #significant positive correlation (p=0.02)
```
```{r prolactin vs pit PRLR, echo = FALSE, message = FALSE}
#correlate plasma prolactin versus pit PRLR exprssion 

#create a datasubset (hyp, prl, join)
p.pr.j<-prl.join%>% filter(focal=="y" & tissue=="pituitary" & geneid=="prlr")

#plot correlation
#(scatterplot for hyp, prl, joined data) = sp.h.pr.j
sp.p.pr.j<- p.pr.j %>% ggplot(aes(y = log.exp, x = Prolactin.ng.mL, color=stage))+
           geom_point()+
          geom_smooth(method=lm, na.rm = TRUE, fullrange= TRUE,
              aes(group=1),colour="black") + 
          labs(x="Plasma prolactin concentration (ng/mL)", 
               y="Pit PRLR expression (log(counts))")+
          ggtitle("Prolactin vs Pit PRLR") + 
          annotate("text", x = 75, y = 3, label = "n.s.")
sp.p.pr.j

#correlation linear model 
corr.p.pr<- lm(log.exp~Prolactin.ng.mL, p.pr.j) #use lm as it gives R2 
summary(corr.p.pr)  #n.s. 
```


```{r crop prolactin correlations, echo=FALSE, message= FALSE}
#import prolactin data (created from a previous join, contains Fred's PRL plasma data)
prolactin<-read.csv("~/Desktop/data/2019-09-11_prolactin_plasma_sample_join.csv")
#join qPCR data and prolactin plasma data 
join<-left_join(crops.long,prolactin)

#scatterplot with linear model graph
join$Stage<-as.factor(join$Stage)
crop_vs_prolactin<- join%>% filter(gene=="PRLR_ddCT") %>%
  ggplot(aes(y = log.foldchange, x = prolactin_conc, color=Stage))+
  geom_point()+
  geom_smooth(method=lm, na.rm = TRUE, fullrange= TRUE,
               aes(group=1),colour="black") + 
  labs(x="Plasma prolactin concentration(ng/mL)", y="Crop PRLR expression (log(fold change))")+
  annotate("text", x = 75, y = -4, label = "p=0.006")+
  ggtitle("Plasma prolactin significantly predicts crop PRLR expression")
crop_vs_prolactin

#prolactin correlation with linear model 
join_PRLR<-join%>% filter(gene=="PRLR_ddCT") #subset joined data
prolactin_corr<- glm(log.foldchange~prolactin_conc, join_PRLR, family="gaussian")
summary(prolactin_corr) 
```

